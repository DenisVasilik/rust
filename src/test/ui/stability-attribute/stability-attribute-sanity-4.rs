// Various checks that stability attributes are used correctly, per RFC 507

#![feature(staged_api)]

#![stable(feature = "rust1", since = "1.0.0")]

mod bogus_attribute_types_2 {
    #[unstable] //~ ERROR attribute must be of the form `#[unstable(...)]`
    fn f1() { }

    #[unstable = "b"] //~ ERROR attribute must be of the form `#[unstable(...)]`
    fn f2() { }

    #[stable] //~ ERROR attribute must be of the form `#[stable(...)]`
    fn f3() { }

    #[stable = "a"] //~ ERROR attribute must be of the form `#[stable(...)]`
    fn f4() { }

    #[stable(feature = "a", since = "b")]
    #[rustc_deprecated] //~ ERROR attribute must be of the form `#[rustc_deprecated(...)]`
    fn f5() { }

    #[stable(feature = "a", since = "b")]
    #[rustc_deprecated = "a"] //~ ERROR attribute must be of the form `#[rustc_deprecated(...)]`
    fn f6() { }
}

fn main() { }
