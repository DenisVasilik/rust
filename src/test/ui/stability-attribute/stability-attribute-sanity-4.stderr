error: attribute must be of the form `#[unstable(...)]`
  --> $DIR/stability-attribute-sanity-4.rs:8:5
   |
LL |     #[unstable] //~ ERROR attribute must be of the form `#[unstable(...)]`
   |     ^^^^^^^^^^^

error: attribute must be of the form `#[unstable(...)]`
  --> $DIR/stability-attribute-sanity-4.rs:11:5
   |
LL |     #[unstable = "b"] //~ ERROR attribute must be of the form `#[unstable(...)]`
   |     ^^^^^^^^^^^^^^^^^

error: attribute must be of the form `#[stable(...)]`
  --> $DIR/stability-attribute-sanity-4.rs:14:5
   |
LL |     #[stable] //~ ERROR attribute must be of the form `#[stable(...)]`
   |     ^^^^^^^^^

error: attribute must be of the form `#[stable(...)]`
  --> $DIR/stability-attribute-sanity-4.rs:17:5
   |
LL |     #[stable = "a"] //~ ERROR attribute must be of the form `#[stable(...)]`
   |     ^^^^^^^^^^^^^^^

error: attribute must be of the form `#[rustc_deprecated(...)]`
  --> $DIR/stability-attribute-sanity-4.rs:21:5
   |
LL |     #[rustc_deprecated] //~ ERROR attribute must be of the form `#[rustc_deprecated(...)]`
   |     ^^^^^^^^^^^^^^^^^^^

error: attribute must be of the form `#[rustc_deprecated(...)]`
  --> $DIR/stability-attribute-sanity-4.rs:25:5
   |
LL |     #[rustc_deprecated = "a"] //~ ERROR attribute must be of the form `#[rustc_deprecated(...)]`
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 6 previous errors

